libyul/backends/evm/ControlFlow.h

```diff
@@ -39,14 +49,19 @@ struct ControlFlow
 		return nullptr;
 	}
 
-	std::string toDot() const
+	std::string toDot(ControlFlowLiveness const* _liveness=nullptr) const
```

Comment:
Why not `std::optional`?

```suggestion
	std::string toDot(std::optional<ControlFlowLiveness> const& _liveness = std::nullopt) const
```

Reply:
Because the `ControlFlowLiveness` is not copyable and/or movable, so to invoke a method that takes an optional of `ControlFlowLiveness`, you'd have to construct it explicitly in the optional already. So this
```
ControlFlowLiveness liveness(controlFlow);
controlFlow.toDot(liveness);
```
doesn't work.

You'd need something like `std::optional<std::reference_wrapper<ControlFlowLiveness const>>` as type - which I think is quite ugly and semantically the same as passing a const pointer :)

SSCCE: https://godbolt.org/z/aabq6WWj1